jupyter_notebook_parser/parser.py

```diff
@@ -1,87 +1,103 @@
 import json
-from typing import List, Dict
+from os.path import splitext
 
 from jupyter_notebook_parser.container import SourceCodeContainer
 
 
 class JupyterNotebookParser:
-    def __init__(self, notebook_filename: str) -> None:
-        if not notebook_filename.endswith('.ipynb'):
-            raise NameError('This file is not a Jupyter notebook')
+    __notebook_corrupted_msg = "This notebook file is possibly corrupted"
 
-        self.notebook_corrupted_msg = 'This notebook file is possibly corrupted'
+    __slots__ = ("notebook_content",)
+
+    @classmethod
+    def open_from_file(cls, filename, *, assert_file_ext=True, encoding=None):
```

Comment:
This breaks backward compatibility.  If you'd like the ability to open from string, please try to do it within `__init__`